File: modules/multiplexer/pane.go
package multiplexer

import "time"

// PaneState represents the lifecycle state of a managed pane.
type PaneState int

const (
	PaneStateCreated PaneState = iota
	PaneStateRunning
	PaneStateExited
	PaneStateError
)

func (s PaneState) String() string {
	switch s {
	case PaneStateCreated:
		return "created"
	case PaneStateRunning:
		return "running"
	case PaneStateExited:
		return "exited"
	case PaneStateError:
		return "error"
	default:
		return "unknown"
	}
}

// PaneInfo holds metadata about a managed terminal pane.
type PaneInfo struct {
	ID        string    // backend-specific pane identifier
	Name      string    // human-readable agent name
	PID       int       // OS process ID of the pane's shell (if available)
	State     PaneState
	CreatedAt time.Time
	Backend   string // "zellij" or "tmux"
}

File: packages/multiplexer/src/pane.ts
/** PaneState represents the lifecycle state of a managed pane. */
export enum PaneState {
  Created = "created",
  Running = "running",
  Exited = "exited",
  Error = "error",
}

/** PaneInfo holds metadata about a managed terminal pane. */
export interface PaneInfo {
  /** Backend-specific pane identifier */
  id: string;
  /** Human-readable agent name */
  name: string;
  /** OS process ID of the pane's shell (if available) */
  pid?: number;
  state: PaneState;
  createdAt: Date;
  /** "zellij" or "tmux" */
  backend: string;
}
